great_expectations/core/usage_statistics/package_dependencies.py

```diff
@@ -0,0 +1,229 @@
+        print(ge_dependencies.get_dev_dependency_names())
+
+    To verify lists are accurate, you can run this file or execute main() from
+    within a cloned GE repository. This will check the existing requirements
+    files against the static lists returned via the methods above in the
+    usage example and raise exceptions if there are discrepancies.
+"""
+import os
+import re
+from typing import List, Set
+
+
+class GEDependencies:
+    """Store and provide dependencies when requested.
+
+    Also acts as a utility to check stored dependencies match our
+    library requirements.
+
+    Attributes: None
+    """
+
+    """This list should be kept in sync with our requirements.txt file."""
+    GE_REQUIRED_DEPENDENCIES: List[str] = [
+        "altair",
+        "Click",
+        "colorama",
+        "cryptography",
+        "dataclasses",
+        "importlib-metadata",
+        "Ipython",
+        "jinja2",
+        "jsonpatch",
+        "jsonschema",
+        "mistune",
+        "nbformat",
+        "numpy",
+        "packaging",
+        "pandas",
+        "pyparsing",
+        "python-dateutil",
+        "pytz",
+        "requests",
+        "ruamel.yaml",
+        "scipy",
+        "termcolor",
+        "tqdm",
+        "typing-extensions",
+        "urllib3",
+        "tzlocal",
+    ]
+
+    """This list should be kept in sync with our requirements-dev*.txt files."""
+    GE_DEV_DEPENDENCIES: List[str] = [
+        "PyMySQL",
+        "azure-identity",
+        "azure-keyvault-secrets",
+        "azure-storage-blob",
+        "black",
+        "boto3",
+        "feather-format",
+        "flake8",
+        "flask",
+        "freezegun",
+        "gcsfs",
+        "google-cloud-secret-manager",
+        "google-cloud-storage",
+        "ipywidgets",
+        "isort",
+        "mistune",
+        "moto",
+        "nbconvert",
+        "openpyxl",
+        "pre-commit",
+        "psycopg2-binary",
+        "pyarrow",
+        "pyathena",
+        "pyfakefs",
+        "pyodbc",
+        "pypd",
+        "pyspark",
+        "pytest",
+        "pytest-benchmark",
+        "pytest-cov",
+        "pytest-order",
+        "pyupgrade",
+        "requirements-parser",
+        "s3fs",
+        "snapshottest",
+        "snowflake-connector-python",
+        "snowflake-sqlalchemy",
+        "sqlalchemy",
+        "sqlalchemy-bigquery",
+        "sqlalchemy-dremio",
+        "sqlalchemy-redshift",
+        "teradatasqlalchemy",
+        "xlrd",
+    ]
+
+    def __init__(self):
+        self._requirements_relative_base_dir = "../../../"
+        self._dev_requirements_prefix: str = "requirements-dev"
+
+    def get_required_dependency_names(self) -> List[str]:
+        """Sorted list of required GE dependencies"""
+        return sorted(self.GE_REQUIRED_DEPENDENCIES)
+
+    def get_dev_dependency_names(self) -> List[str]:
+        """Sorted list of dev GE dependencies"""
+        return sorted(self.GE_DEV_DEPENDENCIES)
+
```

Comment:
Aren't these already sorted?

Reply:
if we need them sorted, we should do it at declaration, not every time the property is invoked

Reply:
They are but manually sorted by hand. I wanted to make this a bit more lenient to someone inserting a package in the wrong order for internal comparison, especially since the case handling of sorted() and requirements.txt are different (sorted puts caps ahead: ```p = ["a", "Ba", "b", "c"]
sorted(p)
Out[8]: ['Ba', 'a', 'b', 'c']```

Reply:
Good point about sorting at declaration, I had it that way in an earlier version. Changing now.